tin/apps/assignments/models.py

```diff
@@ -279,11 +283,23 @@ def grader_log_filename(self):
             else None
         )
 
-    @property
-    def is_quiz(self):
-        if hasattr(self, "quiz"):
-            return self.quiz
-        return False
+    def quiz_open_for_student(self, student):
+        is_teacher = student in self.course.teacher.all()
```

Comment:
```suggestion
        is_teacher = self.course.teacher.filter(pk=student.id).exists()
```
It could probably be sped up to this, at least from my very brief testing.

Reply:
It's not too much of a performance increase (no courses have over 5 teachers), but fair enough.